pandas/plotting/_matplotlib/misc.py

```diff
@@ -246,7 +246,7 @@ def f(t):
             # appropriately. Take a copy of amplitudes as otherwise numpy
             # deletes the element from amplitudes itself.
             coeffs = np.delete(np.copy(amplitudes), 0)
-            coeffs.resize(int((coeffs.size + 1) / 2), 2)
+            coeffs = np.resize(coeffs, (int((coeffs.size + 1) / 2), 2))
```

Comment:
This was throwing an error during CI:
```
____________________ TestDataFramePlots.test_andrews_curves ____________________
[gw0] linux -- Python 3.8.8 /usr/share/miniconda/envs/pandas-dev/bin/python

self = <pandas.tests.plotting.test_misc.TestDataFramePlots object at 0x7f3ae76442b0>
iris =      SepalLength  SepalWidth  PetalLength  PetalWidth            Name
0            5.1         3.5          1.4       ...      2.3  Iris-virginica
149          5.9         3.0          5.1         1.8  Iris-virginica

[150 rows x 5 columns]

    def test_andrews_curves(self, iris):
        from matplotlib import cm
    
        from pandas.plotting import andrews_curves
    
        df = iris
        # Ensure no UserWarning when making plot
        with tm.assert_produces_warning(None):
>           _check_plot_works(andrews_curves, frame=df, class_column="Name")

pandas/tests/plotting/test_misc.py:158: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/tests/plotting/common.py:625: in _check_plot_works
    raise err
pandas/tests/plotting/common.py:618: in _check_plot_works
    for ret in gen_plots(f, fig, **kwargs):
pandas/tests/plotting/common.py:645: in _gen_two_subplots
    yield f(**kwargs)
pandas/plotting/_misc.py:271: in andrews_curves
    return plot_backend.andrews_curves(
pandas/plotting/_matplotlib/misc.py:282: in andrews_curves
    y = f(t)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

t = array([-3.14159265, -3.11001886, -3.07844506, -3.04687127, -3.01529747,
       -2.98372368, -2.95214988, -2.92057608, ...900229,  2.92057608,  2.95214988,  2.98372368,
        3.01529747,  3.04687127,  3.07844506,  3.11001886,  3.14159265])

    def f(t):
        x1 = amplitudes[0]
        result = x1 / np.sqrt(2.0)
    
        # Take the rest of the coefficients and resize them
        # appropriately. Take a copy of amplitudes as otherwise numpy
        # deletes the element from amplitudes itself.
        coeffs = np.delete(np.copy(amplitudes), 0)
>       coeffs.resize(int((coeffs.size + 1) / 2), 2)
E       ValueError: cannot resize an array that references or is referenced
E       by another array in this way.
E       Use the np.resize function or refcheck=False

pandas/plotting/_matplotlib/misc.py:249: ValueError
```